gateway/middleware.go

```diff
@@ -232,10 +236,17 @@ func (t BaseMiddleware) OrgSessionExpiry(orgid string) int64 {
 	cachedVal, found := ExpiryCache.Get(orgid)
 	if !found {
 		// Cache failed attempt
-		t.SetOrgExpiry(orgid, 604800)
-		go t.OrgSession(orgid)
-		t.Logger().Debug("no cached entry found, returning 7 days")
-		return 604800
+		id, _, _ := orgSessionExpiryCache.Do(orgid, func() (interface{}, error) {
```

Comment:
Maybe I don't understand business logic - but it looks like singleflight should run on `ExpiryCache.Get(orgid)` logic?